scheduler/replica_scheduler.go

```diff
@@ -0,0 +1,94 @@
+package scheduler
+
+import (
+	"github.com/rancher/longhorn-manager/datastore"
+	checker "github.com/rancher/longhorn-manager/scheduler/algorithm"
+	"github.com/rancher/longhorn-manager/types"
+	"github.com/rancher/longhorn-manager/util"
+
+	longhorn "github.com/rancher/longhorn-manager/k8s/pkg/apis/longhorn/v1alpha1"
+)
+
+const (
+	// longhornDirectory is the directory going to be bind mounted on the
+	// host to provide storage space to replica data by default
+	longhornDirectory = "/var/lib/rancher/longhorn/"
+)
+
+type ReplicaScheduler struct {
+	ds *datastore.DataStore
+}
+
+func NewReplicaScheduler(
+	ds *datastore.DataStore) *ReplicaScheduler {
+	rcScheduler := &ReplicaScheduler{
+		ds: ds,
+	}
+	return rcScheduler
+}
+
+func (rcs *ReplicaScheduler) ScheduleReplica(replica *longhorn.Replica) error {
+	// only called when replica is starting for the first time
+	if replica.Spec.DesireState == types.InstanceStateRunning && replica.Spec.NodeID == "" {
+		// get replica list without current replica
+		replicas, err := rcs.ds.GetVolumeReplicas(replica.Spec.VolumeName)
+		if err != nil {
+			return err
+		}
+		if _, ok := replicas[replica.Name]; ok {
+			delete(replicas, replica.Name)
+		}
+
+		// get all hosts
+		nodeIPMap, err := rcs.ds.GetManagerNodeIPMap()
+		if err != nil {
+			return nil
+		}
+		rchecker := checker.NewReplicaChecker(nodeIPMap, replicas)
```

Comment:
It looks unnecessary to create another package `checker`. It's only one function call we needed.

Make it simple for now. We can expand if it's necessary.